lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -248,15 +270,26 @@ def db_import(module, host, user, password, db_name, target, all_databases, port
 
 
 def db_create(cursor, db, encoding, collation):
+    if not db:
+        return False
     query_params = dict(enc=encoding, collate=collation)
-    query = ['CREATE DATABASE %s' % mysql_quote_identifier(''.join(db), 'database')]
-    if encoding:
-        query.append("CHARACTER SET %(enc)s")
-    if collation:
-        query.append("COLLATE %(collate)s")
-    query = ' '.join(query)
-    cursor.execute(query, query_params)
-    return True
+    res = 0
+    for each_db in db:
+        query = ['CREATE DATABASE %s' % mysql_quote_identifier(each_db, 'database')]
+        if encoding:
+            query.append("CHARACTER SET %(enc)s")
+        if collation:
+            query.append("COLLATE %(collate)s")
+        query = ' '.join(query)
+        res += cursor.execute(query, query_params)
+    return res == len(db)
```

Comment:
You should change this one to
```suggestion
    return res > 0
```
to avoid the problem. The function `db_create` is called only in two places; in one place, the return value is ignored (import), and in the other (present), you need to know whether *at least one* DB was created, and not whether *all* DBs have been created. So if you return `res > 0` here, `changed` is determined correctly for `present`.

Reply:
fixed